core/src/from_meta.rs

```diff
@@ -471,6 +470,39 @@ from_syn_parse!(syn::TypeTuple);
 from_syn_parse!(syn::Visibility);
 from_syn_parse!(syn::WhereClause);
 
+impl FromMeta for syn::TypePath {
+    /// Supports both quote-wrapped and bare values.
+    fn from_expr(expr: &Expr) -> Result<Self> {
+        match expr {
+            syn::Expr::Path(body) => {
+                if body.attrs.is_empty() {
+                    Ok(syn::TypePath {
+                        qself: body.qself.clone(),
+                        path: body.path.clone(),
+                    })
+                } else {
+                    Err(Error::custom("attributes are not allowed").with_span(body))
+                }
+            }
+            syn::Expr::Lit(expr_lit) => Self::from_value(&expr_lit.lit),
+            syn::Expr::Group(group) => Self::from_expr(&group.expr),
+            _ => Err(Error::unexpected_expr_type(expr)),
+        }
+    }
+
+    fn from_string(value: &str) -> Result<Self> {
+        syn::parse_str(value).map_err(|_| Error::unknown_value(value))
+    }
+
+    fn from_value(value: &::syn::Lit) -> Result<Self> {
```

Comment:
Nit: It looks like `syn::Lit` is already imported at the module level; we don't need this path here.